src/test/java/TestUtils.java

```diff
@@ -93,4 +94,18 @@ public static void setSearchServletRequestParameters(HttpServletRequest request,
     when(request.getParameter("min")).thenReturn(minPrice);
     when(request.getParameter("max")).thenReturn(maxPrice);
   }
+
+  /** Parses a string containing store analytics into a hashmap representation. */
+  public static HashMap<String, Double> parseStoreAnalytics(String str) throws IOException {
+    String stores = str.substring(str.indexOf("{"), str.indexOf("}") + 1);
+    stores = stores.substring(stores.lastIndexOf("{"), stores.indexOf("}") + 1);
```

Comment:
Is there a more precise way to extract the stores? Perhaps we match for "storeAnalytics"? It's difficult to understand the "{" matching.